src/libs/form/components/SelectWrapper.tsx

```diff
@@ -47,16 +58,38 @@ export function SelectWrapper<T extends string | number>({
 
   const handleChange = (value: string) => {
     const returnValue = initialType === 'number' ? Number(value) : value;
+    setSelectedPreset(returnValue as string);
     onChange?.(returnValue as T);
   };
 
+  const hasPresets = presets && presets.length > 0;
+
   return (
     <Select
       data={parsedData as (string | SelectItem)[]}
       value={parsedValue}
       onChange={handleChange}
       defaultValue={parsedDefaultValue}
       rightSection={loading ? <Loader size={16} /> : null}
+      styles={{ label: hasPresets ? { width: '100%', marginBottom: 5 } : undefined }}
+      label={
+        hasPresets ? (
+          <Group spacing={8} position="apart" noWrap>
+            {label}
+            <PresetOptions
+              color="blue"
+              options={presets}
+              value={selectedPreset}
+              onChange={(value) => {
+                setSelectedPreset(value);
+                onChange?.(value as T);
+              }}
+            />
```

Comment:
You could make a `PresetsLabel` component out of this all in all

Reply:
When we start implementing in other input components, then we definitely should 😛 for now I think it's fine the way it is